### PR TITLE
Deprecate len(Signal)

### LINKED ISSUE #412: deprecate `TimeData.__len__`?
## General

* pyfar version: 0.5.1

## Description

We have  `TimeData.__len__` that returns the number of samples of a TimeData/Signal object. I would suggest to deprecate this for the following reasons:

- We also have `TimeData.n_samples`, which is more explicit
- We do not have `FrequencyData.__len__` which is inconsistent
- `len(time_data_instance)` does not equal `len(time_data_instance.time)` which conflicts with general python use of the function.

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Which issue(s) are closed by this pull request?

Closes #412 

### Changes proposed in this pull request:

- deprecate `len(Signal)`
- test deprecation

### CODE REVIEW

File: pyfar/classes/audio.py [hunk 1]
```diff
@@ -823,6 +824,9 @@ def __repr__(self):
     def __len__(self):
         """Length of the object which is the number of samples stored.
         """
+        warnings.warn(
+            ("len(Signal) wil be deprecated in pyfar 0.8.0 "
```
Comment (sikersten) [inline pyfar/classes/audio.py:828]:
```suggestion
            ("len(Signal) will be deprecated in pyfar 0.8.0 "
```

File: tests/test_deprecations.py [hunk 2]
```diff
@@ -178,3 +181,14 @@ def test_pad_zero_modi():
         with pytest.raises(ValueError):
             # remove mode 'before' and 'after' from pyfar 0.8.0!
             pf.dsp.pad_zeros(pf.Signal([1], 44100), 5, mode='before')
+
+
+def test_signal_len():
+    with pytest.warns(PyfarDeprecationWarning,
+                      match=re.escape("len(Signal) wil be deprecated")):
```
Comment (sikersten) [inline tests/test_deprecations.py:188]:
```suggestion
                      match=re.escape("len(Signal) will be deprecated")):
```

**Review by sikersten — APPROVED** [review]

File: pyfar/classes/audio.py (hunk 1 above)
Comment (f-brinkmann) [reply pyfar/classes/audio.py:828]: Wenn ich groß bin, kann ich irgendwann Rechtschreibung...

File: tests/test_deprecations.py (hunk 2 above)
Comment (f-brinkmann) [reply tests/test_deprecations.py]: fixed :)

**Pull request merged**